packages/babel-generator/src/generators/types.js

```diff
@@ -142,15 +142,27 @@ export function StringLiteral(node: Object, parent: Object) {
     return;
   }
 
-  // ensure the output is ASCII-safe
-  const opts = {
-    quotes: t.isJSX(parent) ? "double" : this.format.quotes,
-    wrap: true
-  };
-  if (this.format.jsonCompatibleStrings) {
-    opts.json = true;
+  let val;
+  if (this.format.disableASCIIEncode) {
```

Comment:
Since the code is already depends on `jsesc` anyway, this probably could just map to `jsesc`'s `minimal` option?